apps/passport-server/src/services/issuanceService.ts

```diff
@@ -35,27 +35,38 @@ import { logger } from "../util/logger";
 
 export class IssuanceService {
   private readonly context: ApplicationContext;
-  private readonly rsaPrivateKey: NodeRSA;
-  private readonly exportedPrivateKey: string;
-  private readonly exportedPublicKey: string;
 
-  public constructor(context: ApplicationContext, rsaPrivateKey: NodeRSA) {
+  private readonly eddsaPrivateKey: string;
+  private readonly rsaPrivateKey: NodeRSA;
+  private readonly exportedRSAPrivateKey: string;
+  private readonly exportedRSAPublicKey: string;
+
+  public constructor(
+    context: ApplicationContext,
+    rsaPrivateKey: NodeRSA,
+    eddsaPrivateKey: string
+  ) {
     this.context = context;
     this.rsaPrivateKey = rsaPrivateKey;
-    this.exportedPrivateKey = this.rsaPrivateKey.exportKey("private");
-    this.exportedPublicKey = this.rsaPrivateKey.exportKey("public");
+    this.exportedRSAPrivateKey = this.rsaPrivateKey.exportKey("private");
+    this.exportedRSAPublicKey = this.rsaPrivateKey.exportKey("public");
+    this.eddsaPrivateKey = eddsaPrivateKey;
   }
 
-  public getPublicKey(): string {
-    return this.exportedPublicKey;
+  public getRSAPublicKey(): string {
+    return this.exportedRSAPublicKey;
+  }
+
+  public async getEdDSAPublicKey(): Promise<[string, string]> {
```

Comment:
This doesn't really need to be async as the only function it calls is synchronous.